optaplanner-core/src/test/java/org/optaplanner/core/impl/phase/PhaseLifecyleTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.optaplanner.core.impl.phase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.core.impl.phase.event.PhaseLifecycleListener;
+import org.optaplanner.core.impl.solver.DefaultSolver;
+import org.optaplanner.core.impl.testdata.domain.TestdataEntity;
+import org.optaplanner.core.impl.testdata.domain.TestdataSolution;
+import org.optaplanner.core.impl.testdata.domain.TestdataValue;
+import org.optaplanner.core.impl.testdata.util.PlannerAssert;
+import org.optaplanner.core.impl.testdata.util.PlannerTestUtils;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PhaseLifecyleTest {
+
+    @Mock
+    private PhaseLifecycleListener<TestdataSolution> listener;
```

Comment:
The @Mock is interesting mockito feature, but I 'll remove this for now because:
- Code encapsulation: The listener can be a local variable. Furthermore, if more methods are added and verify() is used to check the number of mock calls, sharing the same mock between tests is error-prone.
- Consistency: If we start using this style, we should try to migrate existing code too - otherwise our code becomes an inconsistent mess - and if such partial introductions of technologies happen a few times, bugs start to appear.
- There can only be one @RunWith annotation: What if we want to write a unit test that runs on arquillian, is a parameterized test and uses a bit of mocking too? We'd need to configure 3 Runners somehow.